code/headers/uarm_swift_pro.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <cstring>
+#include <hardware_usart.hpp>
+#include <hwlib.hpp>
+#include <robot_arm_interface.hpp>
+#include <uarm_gcode_generator.hpp>
+#include <vector3.hpp>
+
+/**
+ * Class uArm_swift_pro provides a implementation for uArm swift pro.
+ */
+namespace r2d2::robot_arm {
+    class uarm_swift_pro_c : public robot_arm_interface_c {
+    protected:
+        const uint16_t default_speed = 1000;
+
+    private:
+        uarm_gcode_generator_c<50> gcode_generator;
+        hardware_usart_c usart_bus;
+
+    public:
+        /**
+         * Robot arm constructor, needs an usart bus for communication
+         */
+        uarm_swift_pro_c(const r2d2::hardware_usart_c &usart_bus);
```

Comment:
The usart library offers a interface to use now instead of the `hardware_usart_c`

Reply:
We are at this moment not able to test this change request. We will inform the next team of this needed change.